### PR TITLE
extract seqexec enums

### PR DESCRIPTION
**Pull request by tpolecat**
I'm going to be doing some janitorial work on the seqexec codebase, to help readability. This just moves the enumerated types from `Model.scala` and into their own source files, in their own package.

### CODE REVIEW

**Comment by swalker2m** [comment]
This seems straightforward and correct to me, but I'll let Javier and Carlos take a look.

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala [hunk 1]
```diff
@@ -319,13 +320,13 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
...
-    case Model.Instrument.GPI   => TrySeq(GPI(systems.gpi))
-    case Model.Instrument.GHOST => TrySeq(GHOST(GHOSTController[IO](GDSClient(GDSClient.alwaysOkClient, uri("http://localhost:8888/xmlrpc"))))) // todo put the controller on systems
+  private def toInstrumentSys(inst: Instrument): TrySeq[InstrumentSystem[IO]] = inst match {
+    case Instrument.F2    => TrySeq(Flamingos2(systems.flamingos2, systems.dhs))
+    case Instrument.GmosS => TrySeq(GmosSouth(systems.gmosSouth, systems.dhs))
+    case Instrument.GmosN => TrySeq(GmosNorth(systems.gmosNorth, systems.dhs))
+    case Instrument.GNIRS => TrySeq(Gnirs(systems.gnirs, systems.dhs))
+    case Instrument.GPI   => TrySeq(GPI(systems.gpi))
+    case Instrument.GHOST => TrySeq(GHOST(GHOSTController[IO](GDSClient(GDSClient.alwaysOkClient, uri("http://localhost:8888/xmlrpc"))))) // todo put the controller on systems
     case _                      => TrySeq.fail(Unexpected(s"Instrument $inst not supported."))
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala]: We should likely align this one

File: modules/seqexec/model/src/main/scala/seqexec/model/enum/Resource.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Show.show(_.label)
+
+  implicit val ordering: Ordering[Resource] =
+    order.toOrdering
+}
+
+sealed abstract class Instrument(ordinal: Int, label: String)
+  extends Resource(ordinal, label)
+
+object Instrument {
```
Comment (cquiroz) [inline modules/seqexec/model/src/main/scala/seqexec/model/enum/Resource.scala]: We should try soon to replace this with `gem.enum.Instrument`

**Comment by cquiroz** [comment]
Looks good to me.

File: modules/seqexec/model/src/main/scala/seqexec/model/enum/Resource.scala (hunk 2 above)
Comment (tpolecat) [reply modules/seqexec/model/src/main/scala/seqexec/model/enum/Resource.scala]: Yeah I'm going to tidy up a bit and then I'll circle back and try to replace stuff that exists in core.

**Pull request merged**